### ISSUE TITLE
Generate web page documentation from jsdoc

### ISSUE THREAD

**Issue by macleodbroad-wf** [almende/vis#3383]
## Issue

Web page documentation lives separately from the code and is extra work to maintain. There have been a number of issues relating to out of date or incorrect documentation.

## Proposed Solution

Port all interesting/descriptive content from the web page into the jsdocs and then generate the web pages from the jsdocs.

## Open Questions

Is there an existing template we can use? (ie one of the ones here: https://github.com/jsdoc3/jsdoc#templates)

Should we create a custom template that matches the existing visjs.org site?

## Note

This issue can be assigned to me, but I'd like some feedback on the open questions before I begin, so I do not waste effort pursuing this.

**Comment by macleodbroad-wf** [comment]
fyi @yotamberk - this may overlap with some of the work being done in https://github.com/almende/vis/issues/3364

**Comment by yotamberk** [comment]
Not necessarily

**Comment by wimrijnders** [comment]
> Should we create a custom template that matches the existing visjs.org site?

I'd prefer this option.

I considered this some time ago, and my personal conclusion was to use something like [EJS](http://www.embeddedjs.com/) for the document generation. My overriding concern was to eliminate redundancy in the documentation, which is present a lot. However, this would not use the jsdoc tags.

A middle ground would be much better. I'm not convinced, though, that the documentation can be fully generated from just jsdoc tags.

**Comment by yotamberk** [comment]
I totally agree with @wimrijnders . We need to match the existing site. 
I'm too concern about how the docs will be generated and if they will really be beneficial to the user.

**Comment by macleodbroad-wf** [comment]
I agree that jsdoc alone is not enough to generate all of the documentation, however, there is a markdown plugin for jsdoc that allows you to generate pages from markdown + jsdoc.

I think we can port the existing site content to jsdoc with the markdown readme, and create a template which renders into the same state as the existing. We can also generate a strict subset, so that the docs generated are only those that are required to cover the user facing API.

The idea is just to remove/reduce the burden of updating documentation in two places (the jsdoc & the website), and rather just update the jsdoc, and have the website updated automatically.

**Comment by wimrijnders** [comment]
That sounds actually very good. Doing 'strict subsets' sounds like this could be implemented incrementally. And I totally agree with the motivation in the last paragraph.

Can you give a link to the plugin you mention?

**Comment by macleodbroad-wf** [comment]
Actually it looks like 3.3.0+ doesn't require the markdown plugin: http://usejsdoc.org/about-including-readme.html
